packages/dd-trace/test/plugins/agent.js

```diff
@@ -41,7 +41,7 @@ module.exports = {
         server = http.createServer(agent)
         server.on('connection', socket => sockets.push(socket))
 
-        listener = server.listen(port, 'localhost', resolve)
+        listener = server.listen(port, 'localhost', () => resolve(server))
```

Comment:
I can't do the `agent.server = server` we mentioned because what we require is this file, not `agent`. I'd have to do add a `getAgent()` function to the `module.exports` in this file, which seems worse. 
We also call `agent` to the result of the `require` of this file, so we'd have to rename that in every file. 

I think this accomplishes the flexibility